Add new buttons docs - first part. (#835)
telerik/maui-docs: controls/togglebutton/visual-states.md

---
title: Visual States
page_title: .NET MAUI ToggleButton Documentation - Visual States

> description: Learn how to set the border color, border thickness and other in different Visual States of the Telerik ToggleButton for .NET MAUI.

Reviewer: ```suggestion
description: Learn how to set the border color, border thickness and other in the different visual states of the Telerik ToggleButton for .NET MAUI.
```

position: 9
tags: .net maui, telerik .net maui, ui for .net maui, toggle, button, microsoft .net maui
slug: togglebutton-visual-states
---

# .NET MAUI ToggleButton Visual States

> The ToggleButton provides the following `CommonStates` visual states described in the table below:

Reviewer: ```suggestion
The ToggleButton provides the following `CommonStates` visual states:
```

> | Visual States | Description |

Reviewer: Apply for all Visual State tables throughout the PR.

```suggestion
| Visual State | Description |
```

| ------------- | --------------- |

> | `Normal` | Applies when the button is in normal state |

Reviewer: Add periods (`.`) at the end of every description in all tables.

| `Pressed` | Applies when the button is pressed |

> | `MouseOver` | Applies when the mouse pointer hovers over the control - applicable for Desktop |

Reviewer: ```suggestion
| `MouseOver` | (Desktop-only) Applies when the mouse pointer hovers over the component. |
```

| `Disabled` | Applies when the button is disabled |

> | `Toggled` | Applies when the button is disabled |

Reviewer: "is toggled" instead of "is disabled"?

Reviewer: Also all descriptions from here on have become "when the button is disabled", so they probably need to be revised.

| `ToggledMouseOver` | Applies when the button is disabled |
| `ToggledPressed` | Applies when the button is disabled |
| `ToggledDisabled` | Applies when the button is disabled |
| `Indeterminate` | Applies when the button is disabled |
| `IndeterminateMouseOver` | Applies when the button is disabled |
| `IndeterminatePressed` | Applies when the button is disabled |
| `IndeterminateDisabled` | Applies when the button is disabled |

> The ToggleButton provides the following `FocusStates` visual states described in the table below:

Reviewer: ```suggestion
The ToggleButton provides the following `FocusStates` visual states:
```

| Visual States | Description |
| ------------- | --------------- |

> | `Unfocused` | Applies when the button is in normal state |

Reviewer: Are these descriptions correct?

| `Focused` | Applies when the button is pressed |

### Using the Visual States

The following example demonstrates how to use the ToggleButton Visual States.

**1.** Define the ToggleButton in XAML:

<snippet id='togglebutton-visual-states' />

> **2.** Define the Visual States in the page's resources:

Reviewer: ```suggestion
**2.** Define the visual states in the page's resources:
```

<snippet id='togglebutton-visual-states-resources' />

**3.** Add the `telerik` namespace:

```XAML
xmlns:telerik="http://schemas.telerik.com/2022/xaml/maui"
```

> > For a runnable example demonstrating the ToggleButton Visual States, see the [SDKBrowser Demo Application]({%slug sdkbrowser-app%}) and go to **ToggleButton > Features** category.

Reviewer: ```suggestion
>tip For a runnable example demonstrating the ToggleButton visual states, see the [SDKBrowser Demo Application]({%slug sdkbrowser-app%}) and go to the **ToggleButton > Features** category.
```

---
title: Visual States
page_title: .NET MAUI ToggleButton Documentation - Visual States
description: Learn how to set the border color, border thickness and other in the different visual states of the Telerik ToggleButton for .NET MAUI.
position: 9
tags: .net maui, telerik .net maui, ui for .net maui, toggle, button, microsoft .net maui
slug: togglebutton-visual-states
---

# .NET MAUI ToggleButton Visual States

The ToggleButton provides the following `CommonStates` visual states:

| Visual State | Description |
| ------------- | --------------- |
| `Normal` | Applies when the button is in normal state. |
| `Pressed` | Applies when the button is pressed. |
| `MouseOver` | (Desktop-only) Applies when the mouse pointer hovers over the control. |
| `Disabled` | Applies when the button is disabled. |
| `Toggled` | Applies when the button is toggled. |
| `ToggledMouseOver` | (Desktop-only) Applies when the button is toggled and the mouse is over the button. |
| `ToggledPressed` | Applies when the button is toggled and pressed. |
| `ToggledDisabled` | Applies when the button is toggled and disabled. |

> | `Indeterminate` | Applies when the button is in indeterminate state. |

Reviewer: ```suggestion
| `Indeterminate` | Applies when the button is in an indeterminate state. |
```

> | `IndeterminateMouseOver` | (Desktop-only) Applies when the button is in indeterminate state and the mouseis over the button. |

Reviewer: ```suggestion
| `IndeterminateMouseOver` | (Desktop-only) Applies when the button is in an indeterminate state and the mouse is over the button. |
```

> | `IndeterminatePressed` | Applies when the button is in indeterminate state and pressed. |

Reviewer: ```suggestion
| `IndeterminatePressed` | Applies when the button is in an indeterminate state and pressed. |
```

> | `IndeterminateDisabled` | Applies when the button is in indeterminate state and disabled. |

Reviewer: ```suggestion
| `IndeterminateDisabled` | Applies when the button is in an indeterminate state and disabled. |
```
